compiler/rustc_borrowck/src/type_check/liveness/trace.rs

```diff
@@ -83,8 +80,8 @@ struct LivenessContext<'a, 'typeck, 'tcx> {
     drop_data: FxIndexMap<Ty<'tcx>, DropData<'tcx>>,
 
     /// Results of dataflow tracking which variables (and paths) have been
-    /// initialized.
-    flow_inits: ResultsCursor<'a, 'tcx, MaybeInitializedPlaces<'a, 'tcx>>,
+    /// initialized. Computed lazily when needed by drop-liveness.
+    flow_inits: Option<ResultsCursor<'a, 'tcx, MaybeInitializedPlaces<'a, 'tcx>>>,
```

Comment:
this could be a `OnceCell` and avoid the need for the few `&mut self`s added in this commit.

https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html#method.get_or_try_init

Reply:
This is used to give the `&mut ResultsCursor` we need, so I'm not seeing how the `&mut self`s could be avoided?

Reply:
It's a `&mut self`.

Reply:
Ah, right - so it wouldn't avoid the `&mut self`. It would just be slightly cleaner, but wouldn't affect the receiver type.